### ISSUE TITLE
no-redeclare and Flow declarations

### ISSUE THREAD

**Issue by haggholm** [babel/eslint-plugin-babel#162]
Flow allows for explicit function declarations. I guess they’re mostly intended for separate declaration files, but I also find them very useful for documenting functions with more than one parameter format. Often, this seems to happen when a function with positional parameters gets too complex and is modified to an options object, but remains compatible with the old params to minimise disruption.

```jsx
declare function foo(bar: boolean, baz: string, callback: Function): void;
declare function foo({ bar: boolean, baz: string, beep: SomeType, bloop: Object }, callback: Function): void;

function foo() {
   if (typeof arguments[0] === 'Object') // ...
}
```

This makes Flow happy and keeps type checking working, *but* it results in lots and lots of eslint warnings from `no-redeclare`.

Based on similar issues, it seems to me that this is the right project to raise this issue—if not, please let me know. It would be marvellous if `eslint` could simply ignore explicit `declare` statements for the purposes of the `no-redeclare` rule. Presumably, the point of `no-redeclare` is to avoid overriding actual runtime bindings.

**Comment by suchipi** [comment]
Seems reasonable. I'm not sure whether this should be fixed in eslint or this plugin, though. I would have to do some research.

**Comment by kaicataldo** [comment]
Thank you for the issue. Now that [@babel/eslint-plugin has been released](https://babeljs.io/blog/2020/07/13/the-state-of-babel-eslint), we are making this repository read-only. If this is a change you would still like to advocate for, please reopen this in the [babel/babel monorepo](https://github.com/babel/babel).
